include/nbl/builtin/hlsl/subgroup2/ballot.hlsl

```diff
@@ -11,6 +11,19 @@ namespace hlsl
 namespace subgroup2
 {
 
+uint32_t LastSubgroupInvocation()
+{
+    // why this code was wrong before:
+    // - only compute can use SubgroupID
```

Comment:
SubgroupID is the subgroup ID, not SubgroupInvocationID

I'd template
```cpp
template<int32_t AssumeAllActive=false>
uint32_t LastSubgroupInvocation()
{
   if (AssumeAllActive)
      return glsl::gl_SubgroupSize()-1;
   else
      return glsl::subgroupBallotFindMSB(glsl::subgroupBallot(true));
}
```

btw in code that knows you're going to have all invocations active and you KNOW the subgroup size, you should never call `LastSubgroupInvocation` or `ElectLast`